synapse/server.py

```diff
@@ -479,7 +489,7 @@ def get_presence_router(self) -> PresenceRouter:
         return PresenceRouter(self)
 
     @cache_in_self
-    def get_typing_handler(self) -> FollowerTypingHandler:
+    def get_typing_handler(self) -> Union[TypingWriterHandler, FollowerTypingHandler]:
```

Comment:
I don't think we need this change, `TypingWriterHandler` is a sub-class of `FollowerTypingHandler`.